provider/helm3/implementer.go

```diff
@@ -0,0 +1,107 @@
+package helm3
+
+import (
+    "os"
+    "strings"
+
+    "helm.sh/helm/v3/pkg/chart"
+
+	log "github.com/sirupsen/logrus"
+
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/release"
+    "helm.sh/helm/v3/pkg/cli"
+)
+
+// to do:
+	// * update to latest chart package
+	// * udpate the paramateres for the function
+
+const DefaultUpdateTimeout = 300
+
+// Implementer - generic helm implementer used to abstract actual implementation
+type Implementer interface {
+	// ListReleases(opts ...helm.ReleaseListOption) ([]*release.Release, error)
+    ListReleases() ([]*release.Release, error)
+	UpdateReleaseFromChart(rlsName string, chart *chart.Chart, vals map[string]string, opts ...bool) (*release.Release, error)
+}
+
+// Helm3Implementer - actual helm3 implementer
+type Helm3Implementer struct {
+	actionConfig *action.Configuration
+}
+
+// NewHelm3Implementer - get new helm implementer
+func NewHelm3Implementer() *Helm3Implementer {
+    settings := cli.New()
+
+    actionConfig := &action.Configuration{}
+    // You can pass an empty string instead of settings.Namespace() to list
+    // all namespaces
+    if err := actionConfig.Init(settings.RESTClientGetter(), settings.Namespace(), os.Getenv("HELM_DRIVER"), log.Printf); err != nil {
```

Comment:
it should be able to either monitor specific namespace or all of them. Also, could we pass all required parameters to the `NewHelm3Implementer()` like `NewHelm3Implementer(options)` and options being:
```
type Options struct {
   Namespace string  // monitor either specific or all of them if empty. 
   Driver           string // from os.Getenv("HELM_DRIVER")
}
```
If we need to create `actionConfig` per namespace then this should probably be moved into a helper function that gets it for a namespace and then used together with Kubernetes `go-client` to loop through namespaces :) Anyway, excellent work

Reply:
oops, thanks didn't catch the fact that its only tracking the namespace its installed to.
Do you think it'd be fine to simply have keel track all namespaces on the cluster? (basically just changing the settings.Namespace() param input to an empty string.) Since its seems like both Helm v2 and k8s provider just track all namespaces also.

So if we only just let it track all namespaces do you think the options struct would still be necessary?

Reply:
yes, tracking all namespaces would be good :) We can add specific namespace tracking support later to both k8s and helm providers